content/en/apps/guides/hosting/core-developer.md

```diff
@@ -0,0 +1,235 @@
+
+```shell
+docker run -d -p 5984:5984 -p 5986:5986 --name medic-couchdb -e COUCHDB_USER=medic -e COUCHDB_PASSWORD=password -v ~/cht-docker/local.d:/opt/couchdb/data -v ~/cht-docker/local.d:/opt/couchdb/etc/local.d apache/couchdb:2
+```
+
+Let's ensure CouchDB is set up with a test `curl` call. This should show "nonode@nohost" in JSON:
+
+```shell
+curl -X GET "http://medic:password@localhost:5984/_membership" | jq
+```
+
+Every time you run any `grunt` or `node` commands, it will expect `COUCH_NODE_NAME` and `COUCH_URL` environment variables to be set:
+
+```shell
+echo "export COUCH_NODE_NAME=nonode@nohost">> ~/.bashrc
+echo "export COUCH_URL=http://medic:password@localhost:5984/medic">> ~/.bashrc
+. ~/.bashrc
+echo $COUCH_NODE_NAME&& echo $COUCH_URL
+```
+
+### CHT Core cloning and setup
+
+Clone the main CHT Core repo from GitHub and initialize it with an `npm` command. Note this last command may take many minutes. Be patient!
+
+```shell
+git clone https://github.com/medic/cht-core ~/cht-core
+cd ~/cht-core
+npm ci  
+```
+
+We need to harden couch with a `grunt` call, required even in development:
+
+```shell
+grunt secure-couchdb
+curl -X PUT "http://medic:password@localhost:5984/_node/$COUCH_NODE_NAME/_config/httpd/WWW-Authenticate" -d '"Basic realm=\"administrator\""' -H "Content-Type: application/json"
+```
+
+### Developing
+
+Now you have everything installed and can begin development! You'll need three separate terminals when doing development. In the first terminal, run:
+
+```shell
+cd ~/cht-core&&grunt                            
+```
+
+Be **very** patient until you see:
+
+> "Waiting..."
+
+Then in a 2nd terminal run:
+
+```shell
+cd ~/cht-core&&grunt dev-api
+```
+
+Finally, in a 3rd terminal run:
+
+```shell
+cd ~/cht-core&&grunt dev-sentinel
+```
+
+That's it!  Now when you edit code in your IDE, it should automatically reload.  You can see the CHT running locally here: [http://localhost:5988/](http://localhost:5988/)
+
+When you're done with development you can `ctrl + c` in the three terminals and stop the CouchDB container with `docker stop medic-couchdb`.  When you want to resume development later, run `docker start medic-couchdb` and re-run the three terminal commands.
+
+## Other Paths
+
+If you weren't able to follow the happy path above, here are some details about the developer install that may help you troubleshoot what went wrong.
+
+### Prerequisites
+
+If you had issues with following the above steps, check out these links for how to install the prerequisites on your specific platform:
+
+* [Node.js 12.x](https://nodejs.org/)
+* [npm 6.x.x](https://npmjs.com/)
+* [grunt cli](https://gruntjs.com/using-the-cli)
+* [xsltproc](http://www.sagehill.net/docbookxsl/InstallingAProcessor.html) 
+* [python 2.7](https://www.python.org/downloads/)
```

Comment:
Gotcha. I dug up [this ticket](https://github.com/medic/cht-core/issues/5608) which says maybe it won't work? I don't want to solve this now given how far reaching it might be.  I say we leave it as is and do nothing or leave it as is and file a "update requirements to be python 3" ticket.

Thoughts?

Reply:
Meh - doing [the latter](https://github.com/medic/cht-core/issues/7636).